lib/kubernetes-deploy/kubernetes_resource/job.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class Job < KubernetesResource
```

Comment:
I wouldn't insist on this, but it could be helpful if you extracted these new class additions into a separate PR. Making sure each one has appropriate checks and test coverage is not really related to this PR.

Reply:
As a prerequisite PR? Without the new classes this PR won't work.